### DISCUSSION TITLE
How to view debug information for flowise-components?

### DISCUSSION THREAD

**Discussion by mattli001** [FlowiseAI/Flowise#3124] [Q&A]
Hello Flowise community,
I'm working on improving the 'analyse chatflow' functionality in Flowise, specifically focusing on the Langfuse integration. I've noticed that in the Langfuse trace records, the "input" content is empty in the angentflow. 

* agentflow
  <img width="1693" alt="截圖 2024-09-01 下午1 02 31" src="https://github.com/user-attachments/assets/639b2f11-b3ea-49a2-a49a-12af090787ff">
* Langfuse session
  <img width="765" alt="截圖 2024-09-01 下午1 03 44" src="https://github.com/user-attachments/assets/dfcea033-9c94-4bb5-9f97-fdc2e4d523ee">

To troubleshoot this issue, I need to enable debugging for the components.

Currently, I know that in the Flowise development environment and view debug information for `flowise.dev` by setting `DEBUG=true` in the `packages/server/.env` file. Ref: [How to view the debug under Docker?](https://github.com/FlowiseAI/Flowise/discussions/758#top)

However, I'm trying to figure out how to view debug information for `flowise-components`. Such as in this code example:
https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/agents/OpenAIAssistant/OpenAIAssistant.ts#L158

<img width="583" alt="截圖 2024-09-01 中午12 14 22" src="https://github.com/user-attachments/assets/03bd4db3-aa02-4fc0-ac23-1dce8ca85b7a">

Is there a similar setting or method that would allow me to see debug messages for these components during the development process?

Any help or guidance would be greatly appreciated!
Thank you in advance for your assistance.

**Comment by auraeinteractive** [comment]
I agree. There's the possibility to add code to a node - therefore, it would help with some debugging.
